src/enclave/App/App.cpp

```diff
@@ -65,107 +68,303 @@
 # define ENCLAVE_FILENAME "enclave.signed.so"
 #endif
 
-static sgx_ra_context_t context = INT_MAX;
 JavaVM* jvm;
 
-typedef struct _sgx_errlist_t {
-  sgx_status_t err;
+typedef struct _oe_errlist_t {
+  oe_result_t err;
   const char *msg;
   const char *sug; /* Suggestion */
-} sgx_errlist_t;
-
-/* Error code returned by sgx_create_enclave */
-static sgx_errlist_t sgx_errlist[] = {
-  {
-    SGX_ERROR_UNEXPECTED,
-    "Unexpected error occurred.",
-    NULL
-  },
-  {
-    SGX_ERROR_INVALID_PARAMETER,
-    "Invalid parameter.",
-    NULL
-  },
-  {
-    SGX_ERROR_OUT_OF_MEMORY,
-    "Out of memory.",
-    NULL
-  },
-  {
-    SGX_ERROR_ENCLAVE_LOST,
-    "Power transition occurred.",
-    "Please refer to the sample \"PowerTransition\" for details."
-  },
-  {
-    SGX_ERROR_INVALID_ENCLAVE,
-    "Invalid enclave image.",
-    NULL
-  },
-  {
-    SGX_ERROR_INVALID_ENCLAVE_ID,
-    "Invalid enclave identification.",
-    NULL
-  },
-  {
-    SGX_ERROR_INVALID_SIGNATURE,
-    "Invalid enclave signature.",
-    NULL
-  },
-  {
-    SGX_ERROR_OUT_OF_EPC,
-    "Out of EPC memory.",
-    NULL
-  },
-  {
-    SGX_ERROR_NO_DEVICE,
-    "Invalid SGX device.",
-    "Please make sure SGX module is enabled in the BIOS, and install SGX driver afterwards."
-  },
-  {
-    SGX_ERROR_MEMORY_MAP_CONFLICT,
-    "Memory map conflicted.",
-    NULL
-  },
-  {
-    SGX_ERROR_INVALID_METADATA,
-    "Invalid enclave metadata.",
-    NULL
-  },
-  {
-    SGX_ERROR_DEVICE_BUSY,
-    "SGX device was busy.",
-    NULL
-  },
-  {
-    SGX_ERROR_INVALID_VERSION,
-    "Enclave version was invalid.",
-    NULL
-  },
-  {
-    SGX_ERROR_INVALID_ATTRIBUTE,
-    "Enclave was not authorized.",
-    NULL
-  },
-  {
-    SGX_ERROR_ENCLAVE_FILE_ACCESS,
-    "Can't open enclave file.",
-    NULL
-  },
+} oe_errlist_t;
+
+/* Error codes defined by oe_result_t */
+static oe_errlist_t oe_errlist[] = {
```

Comment:
How about moving these error strings into a separate file for readability?

Reply:
Unaddressed.

Reply:
Moved to a different file.